batch/batch/driver/main.py

```diff
@@ -1218,33 +1206,26 @@ def set_value(gauge, data):
 def monitor_instances(app) -> None:
     driver: CloudDriver = app['driver']
     inst_coll_manager = driver.inst_coll_manager
+    resource_rates = driver.billing_manager.resource_rates
 
     cost_per_hour: Dict[CostPerHourLabels, List[float]] = defaultdict(list)
     free_cores: Dict[InstCollLabels, List[float]] = defaultdict(list)
-    utilization: Dict[InstCollLabels, List[float]] = defaultdict(list)
+    total_cores: Dict[InstCollLabels, List[float]] = defaultdict(list)
     instances: Dict[InstanceLabels, int] = defaultdict(int)
 
     for inst_coll in inst_coll_manager.name_inst_coll.values():
-        for instance in inst_coll.name_instance.values():
-            # free cores mcpu can be negatively temporarily if the worker is oversubscribed
-            utilized_cores_mcpu = instance.cores_mcpu - max(0, instance.free_cores_mcpu)
+        inst_coll_labels = InstCollLabels(inst_coll=inst_coll.name)
+        cost_per_hour_labels = CostPerHourLabels(measure='actual', inst_coll=inst_coll.name)
+        revenue_per_hour_labels = CostPerHourLabels(measure='billed', inst_coll=inst_coll.name)
 
+        for instance in inst_coll.name_instance.values():
             if instance.state != 'deleted':
-                actual_cost_per_hour_labels = CostPerHourLabels(measure='actual', inst_coll=instance.inst_coll.name)
-                actual_rate = instance.instance_config.actual_cost_per_hour(driver.billing_manager.resource_rates)
-                cost_per_hour[actual_cost_per_hour_labels].append(actual_rate)
-
-                billed_cost_per_hour_labels = CostPerHourLabels(measure='billed', inst_coll=instance.inst_coll.name)
-                billed_rate = instance.instance_config.cost_per_hour_from_cores(
-                    driver.billing_manager.resource_rates, utilized_cores_mcpu
-                )
-                cost_per_hour[billed_cost_per_hour_labels].append(billed_rate)
-
-                inst_coll_labels = InstCollLabels(inst_coll=instance.inst_coll.name)
-                free_cores[inst_coll_labels].append(instance.free_cores_mcpu / 1000)
-                utilization[inst_coll_labels].append(utilized_cores_mcpu / instance.cores_mcpu)
+                free_cores[inst_coll_labels].append(instance.free_cores_mcpu_nonnegative / 1000)
+                total_cores[inst_coll_labels].append(instance.cores_mcpu)
```

Comment:
I think you need to divide by 1000 here.

Reply:
Indeed! Good catch.